openedx/openedx-platform
# fix: allow non-enterprise, saml users to complete registration in authn MFE

## Description

PR #27431 included a condition that was ment to prevent enterprise users with tpa-hint enabled and coming from SAML IdP to be redirected to the authn MFE, as it "doesn't support a co-branded login/register screen with the enterprise customer (e.g., enterprise logo, custom messaging, etc.)." 

However the condition also prevented non-enterprise, SAML users to be redirected to the authn MFE.

## Supporting information

It was discussed in a [slack conversation](https://openedx.slack.com/archives/C04BM6YC7A6/p1732278823330829).

## Testing instructions

To test this use case:

- Do not use enterprise
- Configure and enable SAML authentication
- Make sure that in the Django admin, Third-party authentication › Provider Configuration, "Skip registration form" is unchecked
- Register via the IdP
- Verify that you are redirected to the authn MFE to complete the registration form.

Tests run as part of PR #27431 should also pass:
Enterprise customer with tpa-hint and SAML IdP should be redirected to the legacy registration page.

## Deadline

ASAP.

## Other information

We happen to have a client that requested SAML authentication, and a number of customizations in the authn MFE including custom fields. Users are then redirected to the [legacy register page](https://github.com/openedx/edx-platform/blob/master/lms/static/js/student_account/views/RegisterView.js), which is based on Backbone. These pages are not theme-able, so porting the customizations is extremely complex and not scalable.
